Dynatrace/dynatrace-operator
# Implements copying codeModules from image.

# Description
This change gets the image specified in the `codeModulesImage` field and unpacks it's contents into the right place using the csi-driver. The outcome is meant to be the same as getting the zip from the tenant, but this way the user can scan the contents of the image they want us to install.

Some extra logic had to be added where the version is important (creating the symlink) as the image not necessarily has this information plainly available.


## How can this be tested?
1. Create an image with the codeModules inside that you want to use. 
2. Set the `codeModulesImage` to that image. 
    - make sure that the operator can "pull" that image, use a custom pull-secret if needed
    - **WARNING**: if you use a custom-pull secret, but still want to get the (host) oneagent images from your tenant, you must have the necessary creds in your custom-pull secret to access the tenant, and cloudNative injection will fail if you don't have running oneagents in your cluster.  (Using applicationMonitoring  with csi-driver for testing is simpler 😉 .)
3.  Deploy the dynakube and some sample apps, and make sure everything works as expected.
    - You can exec into the pod of the csi-driver and check if everything was downloaded properly. (`/data/<tenant>/bin/<image-tag>`)


## Checklist
